### PR TITLE
fix: SDXL LoRA's with diffusion_model keys not working

### PR DESCRIPTION
**Pull request by blessedcoolant**
## Summary

LoRA's with keys starting with `diffusion_model_` were not being parsed. This should hopefully fix it. But I'd like @lstein to take a look at this coz he's more familiar with the Diffusers LoRA key family.

## Testing

LoRA to test: https://civitai.com/models/229213/extremely-detailed-no-trigger-slidersntcaixyz 

1. With PR, works.
2. Without PR, doesn't work.

### CODE REVIEW

**Review by lstein — APPROVED** [review]
No issues spotted and it seems to work!

File: invokeai/backend/lora.py [hunk 1]
```diff
@@ -443,9 +445,12 @@ def _convert_sdxl_keys_to_diffusers_format(cls, state_dict: Dict[str, torch.Tens
                     new_state_dict[full_key] = value
                     not_converted_count += 1
             elif full_key.startswith("lora_te1_") or full_key.startswith("lora_te2_"):
-                # The CLIP text encoders have the same keys in both Stability AI and diffusers formats.
+                # The CLIP text encoders have the same keys in both Stability AI and diffusers formats
                 new_state_dict[full_key] = value
                 continue
+            elif full_key.startswith("diffusion_model_"):
+                new_key = full_key.replace("diffusion_model_", "")
+                new_state_dict[new_key] = value
```
Comment (RyanJDick) [inline invokeai/backend/lora.py:453]: I don't think this is doing what you want. By simply removing the prefix, these keys will silently be skipped here: https://github.com/invoke-ai/InvokeAI/blob/caa7c0f2bd3748980e251f7c506c3d3592e47d3d/invokeai/backend/model_patcher.py#L117-L118

If the goal is to skip these keys, then we should track how many of them got skipped and log a warning.

**Review by RyanJDick — CHANGES_REQUESTED** [review]
This is a weird LoRA. Any idea how it was trained?

Inspecting the state_dict:
```bash
len([k for k in state_dict.keys() if k.startswith("diffusion_model_")])
2
len([k for k in state_dict.keys() if k.startswith("lora_unet_")])
346
len([k for k in state_dict.keys() if k.startswith("lora_te1_")])
48
len([k for k in state_dict.keys() if k.startswith("lora_te2_")])
128
```

I don't know why those two keys have the `diffusion_model_` prefix. To handle it properly, we need to figure out which layers those are supposed to refer to, but I don't really want to handle it if this is a one-off model.

Alternatively, we could skip those keys and log a warning. The LoRA would probably mostly work as expected since it's only 2  keys that are affected.

File: invokeai/backend/lora.py (hunk 1 above)
Comment (blessedcoolant) [reply invokeai/backend/lora.py:453]: This is the only model I've found to have this key so far. You're right. It's probably just better to skip them given its just two layers. I'll close this PR for the time being an reopen another if I come across more LoRA's of this type.

**Pull request closed without merging**